js/main.js

```diff
@@ -62,10 +62,21 @@ function switchView(view) {
   closeNavMenu();
 }
 
-function clickNavList(event) {
+function clickNavLink(event) {
   if (event.target.tagName !== 'A') return;
 
-  switchView(event.target.getAttribute('data-view'));
+  var view = event.target.getAttribute('data-view');
+  if (view === 'favorites') {
+    showFavorites(view);
+  } else { switchView(view); }
+}
+
+function showFavorites(view) {
+  destroyChildren($recipeListContainer);
+  updatePageHeader(view);
+  generateRecipeList(data.favorites);
+  switchView('recipe-list');
+  data.view = 'favorites';
 }
```

Comment:
Is there a reason this needed to be using the same container as the recipe list for searches? If you make this its own separate view, you shouldn't have to add special behavior to your `switchView` function